core/src/main/java/org/infinispan/factories/EmptyConstructorNamedCacheFactory.java

```diff
@@ -113,9 +115,12 @@ public Object construct(String componentName) {
       } else if (componentName.equals(CommandsFactory.class.getName())) {
          return new CommandsFactoryImpl();
       } else if (componentName.equals(PersistenceManager.class.getName())) {
-         PersistenceManagerImpl persistenceManager = new PersistenceManagerImpl();
+         PersistenceManager persistenceManager = new PersistenceManagerImpl();
          if (configuration.persistence().passivation()) {
-            return new PassivationPersistenceManager(persistenceManager);
+            persistenceManager = new PassivationPersistenceManager(persistenceManager);
+         }
+         if (configuration.tracing().enabled(SpanCategory.PERSISTENCE)) {
```

Comment:
```suggestion
   if (globalConfiguration.tracing().enabled()) {
```
As I mentioned in my other comment component replacement like this can not be rewired at runtime currently, so it must pay attention to the global configuration that is not runtime enablable.

Reply:
you're definitively right! fixing it. thanks!

Reply:
fixed thanks